src/mqueryfront/src/query/QueryMatches.js

```diff
@@ -9,24 +9,48 @@ import { PT_MATCHES, PT_PAGINATION } from "../queryUtils";
 const QueryMatches = (props) => {
     const { matches, qhash, pagination } = props;
 
-    const matchesList = matches.map((match, index) => {
-        const download_url =
-            API_URL +
-            "/download?job_id=" +
-            encodeURIComponent(qhash) +
-            "&ordinal=" +
-            encodeURIComponent(index) +
-            "&file_path=" +
-            encodeURIComponent(match.file);
+    const [filters, setFilter] = useState([]);
 
-        return (
-            <QueryMatchesItem
-                key={match.file}
-                match={match}
-                download_url={download_url}
-            />
-        );
-    });
+    const updateFilter = (name) => {
+        if (!filters.includes(name)) {
+            setFilter([...filters, name]);
+        } else {
+            setFilter(filters.filter((e) => e !== name));
+        }
+    };
+
+    const matchesList = matches
+        .filter((match) => {
+            if (
+                filters.length > 0 &&
+                match.matches.some((v) => filters.includes(v))
+            ) {
+                return match;
+            } else {
+                return match;
```

Comment:
Both branches return the same thing, so this all is a no-op. Is this `.filter` necessary?